plugins/modules/katello_activation_key.py

```diff
@@ -145,88 +114,151 @@ def override_to_boolnone(override):
+    content_overrides = entity_dict.pop('content_overrides', None)
+    host_collections = entity_dict.pop('host_collections', None)
+    changed, activation_key = module.ensure_entity('activation_keys', entity_dict, entity, params=scope)
 
+    try:
         # only update subscriptions of newly created or updated AKs
         # copied keys inherit the subscriptions of the origin, so one would not have to specify them again
         # deleted keys don't need subscriptions anymore either
-        if state == 'present' or (state == 'present_with_defaults' and changed):
-            if 'subscriptions' in entity_dict:
-                subscriptions = entity_dict['subscriptions']
-                desired_subscription_ids = set(s.id for s in find_subscriptions(module, subscriptions, entity_dict['organization']))
-                current_subscriptions = [Subscription(**result)
-                                         for result in Subscription().search_normalize(activation_key_entity.subscriptions()['results'])]
-                current_subscription_ids = set(s.id for s in current_subscriptions)
+        if module.state == 'present' or (module.state == 'present_with_defaults' and changed):
+            ak_scope = {'activation_key_id': activation_key['id']}
+            # ak_scope.update(scope)  # See Hack above.
+            if subscriptions is not None:
+                subscription_names = [subscription['name'] for subscription in subscriptions]
+                desired_subscriptions = module.find_resources_by_name('subscriptions', subscription_names, params=scope, thin=True)
+                desired_subscription_ids = set(item['id'] for item in desired_subscriptions)
+                current_subscriptions = module.list_resource('subscriptions', params=ak_scope)
+                current_subscription_ids = set(item['id'] for item in current_subscriptions)
 
                 if desired_subscription_ids != current_subscription_ids:
                     if not module.check_mode:
-                        for subscription_id in (desired_subscription_ids - current_subscription_ids):
-                            activation_key_entity.add_subscriptions(data={'quantity': 1, 'subscription_id': subscription_id})
-                        for subscription_id in (current_subscription_ids - desired_subscription_ids):
-                            activation_key_entity.remove_subscriptions(data={'subscription_id': subscription_id})
+                        ids_to_remove = current_subscription_ids - desired_subscription_ids
+                        if ids_to_remove:
+                            payload = {
+                                'id': activation_key['id'],
+                                'subscriptions': [{'id': item} for item in ids_to_remove],
+                            }
+                            payload.update(scope)
+                            module._resource_action('activation_keys', 'remove_subscriptions', payload)
```

Comment:
in #392 I made `resource_action` fully public, as it felt weird to call `_resource_action` on anything but `self`. what do you think? or @ekohl?

Reply:
I think it's not yet merged, but i like it.